### ISSUE TITLE
Exasol Performance Degradation

### ISSUE THREAD

**Issue by wade-cole** [dbeaver/dbeaver#8889]
#### System information: 
- Operating system (distribution) and version Windows 10 Enterprise Version 1909
- DBeaver version 7.1.0.202005311732
- Additional extensions

#### Connection specification:
- Database name and version Exasol 6.1.8
- Driver name Exasol JDBC Driver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
I update my DBeaver version religiously upon every release. Starting in version 7.0.5 and persisting in version 7.1.0, certain operations take excessively long to resolve. For example, often when I query a view with a query that should return immediately, I instead have to wait about a minute while DBeaver is stuck in the event "LoadExasolTableCache". Example query that should be immediate:
SELECT 1 FROM NAM_PROC.STG_ALIGHT_MEDIA LIMIT 1
;
Also often in the Database Navigator when I open a Table tree, DBeaver spins for a minute or so before displaying the tables.
These performance issues didn't exist in prior DBeaver versions.
Thank you for checking this.

**Comment by serge-rider** [comment]
Table list load seems to be more expensive now. Thanks for the bug report

**Comment by serge-rider** [comment]
Possibly because of `/*snapshot execution*/` hint in all metadata queries.
@wade-cole Could you compare these queries performance (set SCHEMA_NAME to some real schema name):
```sql
/*snapshot execution*/ SELECT * FROM "$ODBCJDBC".ALL_TABLES 
WHERE TABLE_SCHEM = 'SCHEMA_NAME' and TABLE_TYPE = 'TABLE' order by TABLE_NAME
```
and
```sql
SELECT * FROM "$ODBCJDBC".ALL_TABLES 
WHERE TABLE_SCHEM = 'SCHEMA_NAME' and TABLE_TYPE = 'TABLE' order by TABLE_NAME
```

**Comment by wade-cole** [comment]
Hi Serge,

Well, in executing those 2 queries, I didn’t get the long wait time. But the one with the hint was slightly longer.

The execution with the hint ran in 2.67 seconds
The execution witOUT the hint ran in 918ms

Let me know if you want me to try anything else. And thank you for your support!

Best Regards,

Wade Cole

adidas
Senior Engineer, Global Operations
T: +1.971.234.7971 | M: +1.360.930.2823
wade.cole@adidas.com<mailto:wade.cole@adidas.com>

**Comment by Sargul** [comment]
this is related to and should be fixed by #9029 and #9031 

I've created pull request #9032 for these issues

**Comment by Sargul** [comment]
@wade-cole the hint has to stay there because the only impact for this hint is that no locks are aquired for the read and meta data query times vary highly in Exasol.

Meta data queries are really the only thing that is very slow in Exasol ;-)
